crates/astria-core/src/primitive/v1/asset/denom.rs

```diff
@@ -742,4 +862,76 @@ mod tests {
         let denom = denom_str.parse::<Denom>().unwrap();
         assert_eq!(denom_str.len(), denom.display_len());
     }
+
+    #[test]
+    fn ibc_prefixed_ord_matches_lexical_sort() {
```

Comment:
I'd like to have these be more atomic and not sort a vec (it's pretty difficult to read).

Instead do something like:

```rust
#[track_caller]
fn assert_denom_order<T: FromStr>(left: &str, right: &str]) {
    let left_denom: T = left.parse().unwrap();
    let right_denom: T = right.parse().unwrap();
    assert_eq!(left_denom.cmp(&right_denom), left.cmp(right));
}

#[test]
fn denoms_are_lexically_ordered() {
    assert_denom_order::<Denom>("ibc/etc", "trace/etc");
    todo!("and all the rest");
}
```

Reply:
Did something similar in 0e5eeb5, let me know if it works for you!